hooks/tk-multi-publish2/basic/nuke_update_flame_clip.py

```diff
@@ -0,0 +1,618 @@
+
+        A dictionary on the following form::
+
+            {
+                "Settings Name": {
+                    "type": "settings_type",
+                    "default": "default_value",
+                    "description": "One line description of the setting"
+            }
+
+        The type string should be one of the data types that toolkit accepts as
+        part of its environment configuration.
+        """
+
+        # settings specific to this class
+        return {
+            "Flame Clip Template": {
+                "type": "template",
+                "default": None,
+                "description": "Template path for flame shot clip path."
+            }
+        }
+
+    @property
+    def item_filters(self):
+        """
+        List of item types that this plugin is interested in.
+
+        Only items matching entries in this list will be presented to the
+        accept() method. Strings can contain glob patters such as *, for example
+        ["maya.*", "file.maya"]
+        """
+        return ["*.sequence"]
+
+    def accept(self, settings, item):
+        """
+        Method called by the publisher to determine if an item is of any
+        interest to this plugin. Only items matching the filters defined via the
+        item_filters property will be presented to this method.
+
+        A publish task will be generated for each item accepted here. Returns a
+        dictionary with the following booleans:
+
+            - accepted: Indicates if the plugin is interested in this value at
+                all. Required.
+            - enabled: If True, the plugin will be enabled in the UI, otherwise
+                it will be disabled. Optional, True by default.
+            - visible: If True, the plugin will be visible in the UI, otherwise
+                it will be hidden. Optional, True by default.
+            - checked: If True, the plugin will be checked in the UI, otherwise
+                it will be unchecked. Optional, True by default.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+
+        :returns: dictionary with boolean keys accepted, required and enabled
+        """
+
+        publisher = self.parent
+        accepted = True
+
+        # ---- Ensure the write node app is configured
+
+        sg_writenode_app = publisher.engine.apps.get("tk-nuke-writenode")
+        if sg_writenode_app:
+            self.logger.debug("Nuke write node app is installed")
+            item.properties["sg_writenode_app"] = sg_writenode_app
+        else:
+            self.logger.debug(
+                "The nuke write node app is not configured for this "
+                "environment. It is required for publishing for Flame."
+                "Plugin %s no accepting item: %s" % (self, item)
+            )
+            accepted = False
```

Comment:
wouldn't it be easier to just return early in all these cases?

Reply:
Yeah, I should change this too.